### PR TITLE
Split up search api

### LINKED ISSUE #125: Add total search counts to search result
Can you please include a total field in the /search/entities/ api?
This `total` tells us about the total available search results. Because the response has only a limited size (e.g. 20)

### PR DESCRIPTION
**Pull request by Henni**
fixes #125 #126 #127 

now the following endpoints support searching and pagination:
```
localhost:10010/artists/?q=i&limit=3&offset=4
localhost:10010/works/
localhost:10010/releases/
localhost:10010/instruments/
```
All these API endpoints create the following response:
```
{
  items: [Artist/Work/Release/Instrument],
  count: integer
}
```

### CODE REVIEW

**Comment by Henni** [comment]
now travis should be happy.

File: api/dsap/search.js [hunk 1]
```diff
@@ -2,16 +2,26 @@
 module.exports = function (context) {
     const models =  context.models;
 
-    function searchArtists(searchString, options, relation) {
-        return models.artists.findAll({
+    function searchArtists(searchString, options) {
+        if (searchString != '') {
+            searchString = '%' + searchString + '%';
+        } else {
```
Comment (pfent) [inline api/dsap/search.js]: This is a little bit an odd one out. Why does this have other semantics and a different query then let's say `findAllReleases`?

**Review by kordianbruck — APPROVED** [review]
Apart from @pfent comment this looks good!

**Review by sacdallago — APPROVED** [review]

File: api/dsap/search.js (hunk 1 above)
Comment (Henni) [reply api/dsap/search.js]: Ah, seems like I didn't commit this file on my second commit. I'll fix it.

**Comment by Henni** [comment]
I reverted all changes to `dsap/search.js` as it should be removed.
I'll do this in a separate PR.

Ready to merge as soon as travis is happy.

**Pull request merged**